### PR TITLE
feat: use sql for show jobs and show job command

### LINKED ISSUE #1446: Fail to get correct result for SHOW JOBS command
wecom-temp-75c35c2f9f7f9ff3d67e55185e181735

### PR DESCRIPTION
**Pull request by tobegit3hub**
* Resolve https://github.com/4paradigm/OpenMLDB/issues/1446 .

### CODE REVIEW

File: src/sdk/sql_cluster_router.cc [hunk 1]
```diff
@@ -1634,14 +1634,15 @@ std::shared_ptr<hybridse::sdk::ResultSet> SQLClusterRouter::HandleSQLCmd(const h
...
-            std::vector<std::vector<std::string>> result;
-            std::vector<std::string> vec = {ss.str()};
-            result.emplace_back(std::move(vec));
-            return ResultSetSQL::MakeResultSet({FORMAT_STRING_KEY}, result, status);
+            std::string db = "__INTERNAL_DB";
+            std::string sql = "SELECT * FROM JOB_INFO";
+            ::hybridse::sdk::Status status;
+            auto rs = ExecuteSQLParameterized(db, sql, std::shared_ptr<openmldb::sdk::SQLRequestRow>(), &status);
+            if (status.code != 0) {
+                std::cout << "ERROR: " << status.msg << std::endl;
```
Comment (dl239) [inline src/sdk/sql_cluster_router.cc]: error msg in `status` will be printed in cmd/sql_cmd.h.

File: src/sdk/sql_cluster_router.cc [hunk 2]
```diff
@@ -1634,14 +1634,15 @@ std::shared_ptr<hybridse::sdk::ResultSet> SQLClusterRouter::HandleSQLCmd(const h
...
-            ShowJobs(false, job_infos);
-            std::stringstream ss;
-            ::openmldb::cmd::PrintJobInfos(job_infos, ss);
-            std::vector<std::vector<std::string>> result;
-            std::vector<std::string> vec = {ss.str()};
-            result.emplace_back(std::move(vec));
-            return ResultSetSQL::MakeResultSet({FORMAT_STRING_KEY}, result, status);
+            std::string db = "__INTERNAL_DB";
+            std::string sql = "SELECT * FROM JOB_INFO";
+            ::hybridse::sdk::Status status;
```
Comment (dl239) [inline src/sdk/sql_cluster_router.cc]: use the status passed by function parameter

File: src/sdk/sql_cluster_router.cc [hunk 3]
```diff
@@ -1654,19 +1653,20 @@ std::shared_ptr<hybridse::sdk::ResultSet> SQLClusterRouter::HandleSQLCmd(const h
...
-            ShowJob(job_id, job_info);
-            std::vector<::openmldb::taskmanager::JobInfo> job_infos;
+            std::string db = "__INTERNAL_DB";
+            std::string sql = "SELECT * FROM JOB_INFO WHERE id = " + std::to_string(job_id);
 
-            if (job_info.id() > 0) {
-                job_infos.push_back(job_info);
+            auto rs = ExecuteSQLParameterized(db, sql, std::shared_ptr<openmldb::sdk::SQLRequestRow>(), status);
+            if (status->code != 0) {
+                std::cout << "ERROR: " << status->msg << std::endl;
```
Comment (tobegit3hub) [inline src/sdk/sql_cluster_router.cc]: Update.

**Comment by tobegit3hub** [comment]
Do not use sql concat.

**Review by vagetablechicken — APPROVED** [review]

**Review by dl239 — APPROVED** [review]

**Pull request merged**